src/Microsoft.AspNet.Mvc.Core/Rendering/Html/TemplateRenderer.cs

```diff
@@ -167,36 +170,38 @@ private IEnumerable<string> GetViewNames()
                 }
 
                 yield return "String";
+                yield break;
             }
-            else if (fieldType.IsInterface())
-            {
-                if (typeof(IEnumerable).IsAssignableFrom(fieldType))
-                {
-                    yield return "Collection";
-                }
-
-                yield return "Object";
-            }
-            else
+            else if (!fieldType.IsInterface())
             {
-                var isEnumerable = typeof(IEnumerable).IsAssignableFrom(fieldType);
-
+                var type = fieldType;
                 while (true)
                 {
-                    fieldType = fieldType.BaseType();
-                    if (fieldType == null)
+                    type = type.BaseType();
+                    if (type == null || type == typeof(Object))
                     {
                         break;
                     }
 
-                    if (isEnumerable && fieldType == typeof(Object))
-                    {
-                        yield return "Collection";
-                    }
+                    yield return type.Name;
+                }
+            }
 
-                    yield return fieldType.Name;
+            if (typeof(IEnumerable).IsAssignableFrom(fieldType))
+            {
+                if (typeof(IEnumerable<IFormFile>).IsAssignableFrom(fieldType))
+                {
+                    yield return nameof(IFormFileCollection);
                 }
+
+                yield return "Collection";
+            }
+            else if (typeof(IFormFile).IsAssignableFrom(fieldType))
+            {
+                yield return nameof(IFormFile);
```

Comment:
if the property is of type `IFormFile`, this will lead to a duplicate entry in the returned collection.  think the only duplication today occurs when the property is of type `object`.  admittedly at the code of complicating the conditions (e.g. `typeof(IFormFile) != fieldType && typeof(IFormFile).IsAssignableFrom(fieldType)`), worth avoiding the duplicate here too.  see for example the special case for `string`, precisely to avoid a duplicate entry.

after fixing the return string for `IEnumerable<IFormFile>`, will also need to deal with potential duplication when `typeof(IEnumerable<IFormFile>) == fieldType`